packages/webapp/src/containers/Finances/useCropSaleInputs.jsx

```diff
@@ -0,0 +1,329 @@
+ * @returns {Object[]} An array of crop variety filter options with unique crop variety IDs and translated labels.
+ * Each option object contains:
+ * - label: The translated label for the crop variety, formatted as "Crop Variety Name, Translated Crop Name".
+ * - value: The unique identifier for the crop variety.
+ */
+const getCropVarietyFilterOptions = (managementPlans, t) => {
+  if (!managementPlans || managementPlans.length === 0) {
+    return [];
+  }
+
+  let cropVarietyFilterOptions = [];
+  let cropVarietySet = new Set();
+
+  for (let mp of managementPlans) {
+    if (!cropVarietySet.has(mp.crop_variety_id)) {
+      cropVarietyFilterOptions.push({
+        label: mp.crop_variety_name
+          ? `${mp.crop_variety_name}, ${t(`crop:${mp.crop_translation_key}`)}`
+          : t(`crop:${mp.crop_translation_key}`),
+        value: mp.crop_variety_id,
+      });
+      cropVarietySet.add(mp.crop_variety_id);
+    }
+  }
+
+  cropVarietyFilterOptions.sort((a, b) => {
+    if (a.label > b.label) {
+      return 1;
+    } else if (b.label > a.label) {
+      return -1;
+    } else {
+      return 0;
+    }
+  });
+
+  return cropVarietyFilterOptions;
+};
+
+/**
+ * Get input components and unregister inactive options using provided data and functions.
+ *
+ * This function generates an array of input components for active options and unregisters
+ * form fields related to inactive options using the provided management plans, system,
+ * currency, React Hook Form functions, disabled input flag, active options, and inactive options.
+ *
+ * @param {Object[]} managementPlans - An array of management plans containing crop variety data.
+ * @param {string} managementPlans[].crop_variety_id - The unique identifier for the crop variety.
+ * @param {Object} system - The system configuration object.
+ * @param {string} currency - The currency used for sale values.
+ * @param {Object} reactHookFormFunctions - Functions provided by React Hook Form library for form management.
+ * @param {boolean} disabledInput - A flag indicating whether the input fields should be disabled.
+ * @param {Object[]} activeOptions - An array of active options for input components.
+ * @param {Object[]} inactiveOptions - An array of inactive options for unregistering form fields.
+ * @returns {React.Component[]} An array of input components for active options.
+ */
+const getInputs = (
+  managementPlans,
+  system,
+  currency,
+  reactHookFormFunctions,
+  disabledInput,
+  activeOptions,
+  inactiveOptions,
+) => {
+  inactiveOptions?.forEach((option) => {
+    const optionRegisterNames = {
+      CROP_VARIETY_ID: `${CROP_VARIETY_SALE}.${option.value}.${CROP_VARIETY_ID}`,
+      SALE_VALUE: `${CROP_VARIETY_SALE}.${option.value}.${SALE_VALUE}`,
+      QUANTITY: `${CROP_VARIETY_SALE}.${option.value}.${QUANTITY}`,
+      QUANTITY_UNIT: `${CROP_VARIETY_SALE}.${option.value}.${QUANTITY_UNIT}`,
+      CHOSEN_VARIETY: `${CROP_VARIETY_SALE}.${option.value}`,
+    };
```

Comment:
I think this can be an array.
```
[
  `${CROP_VARIETY_SALE}.${option.value}.${CROP_VARIETY_ID}`,
  ``${CROP_VARIETY_SALE}.${option.value}.${SALE_VALUE}`,
  ...
]
```

Reply:
Nice! I stopped short of putting it directly into the function so its clear what they are.